pkg/cpio/newc.go

```diff
@@ -160,14 +160,37 @@ func (w *writer) WriteRecord(f Record) error {
 
 	// Write file contents.
 	m, err := io.Copy(w, uio.Reader(f))
+	// An odd wrinkle: in standard cpio and tar, if you get this far, then
+	// an error is not really an error. It's considered a short read.
```

Comment:
Even permission errors? Like can't open that file? I feel like that kinda stuff we can't suppress for this niche use case. For the niche, a flag?

Reply:
no, this should work like cpio and tar do, and that requires no flag. And, in linux, there are some files where access(2) passes but a read gets an einval. I want us to act like them.

Reply:
You can make the command pass the flag always, but this package doesn't have to do what cpio and tar do in GNU form.

And that's exactly my point re using access. What's the end goal here?

Reply:
the goal is for me to be able to cpio up /proc and /sys and other such things that are not quite files but close enough.

I have a real world use case. I can fulfill that case by just using GNU cpio and leaving u-root cpio broken, if that is the desired way forward.

Reply:
I don't understand why we would cpio up proc or sys.

You're gonna create /proc, and /proc/whatever, and then mount procfs over /proc anyway, hiding all the files in the cpio?

Reply:
I'm not against this, but I want to understand why that would make any sense to do.